### ISSUE TITLE
} java.lang.IllegalStateException: Insufficient spendable states identified for 0.01 TokenType(tokenIdentifier='MXN', fractionDigits=2)

### ISSUE THREAD

**Issue by jarturocastan** [corda/token-sdk#166]
Hi,when trying to use open source 4,1, and 4.3 version, using both token-sdk-1.0 and the latest token-sdk-1.1. we would get very, very slow  at best 2 tx/s per node. We suspect the token-sdk is also giving us problems.  Functionally it works fine, but when trying to stress it out, we noticed  that it was very slow with logs showing this:

[WARN ] 2019-12-09T02:00:20,888Z [Node thread-1] selection.TokenSelection. - TokenType selection failed on attempt 7. {actor_id=user1, actor_owning_identity=O=Banxico, L=London, C=GB, actor_store_id=NODE_CONFIG, fiber-id=10000008, flow-id=0c4f209b-84cc-47a6-9371-00b2e65dbf9c, invocation_id=f56a6cc9-580c-
4076-bb14-b5b3942d102b, invocation_timestamp=2019-12-09T02:00:13.770Z, origin=user1, session_id=e9b0bd3d-599a-474c-abf0-a2d557671d09, session_timestamp=2019-12-09T01:54:07.346Z, thread-id=145}

[WARN ] 2019-12-09T02:00:32,480Z [Node thread-1] selection.TokenSelection. - TokenType selection failed on attempt 8. {actor_id=user1, actor_owning_identity=O=Banxico, L=London, C=GB, actor_store_id=NODE_CONFIG, fiber-id=10000017, flow-id=10bb4d7a-079f-44ba-b22b-df826fc7623f, invocation_id=398ae9f8-598e-45a0-b1b0-29e48f14394d, invocation_timestamp=2019-12-09T02:00:23.645Z, origin=user1, session_id=e9b0bd3d-599a-474c-abf0-a2d557671d09, session_timestamp=2019-12-09T01:54:07.346Z, thread-id=145}

[WARN ] 2019-12-09T03:33:21,400Z [Node thread-1] selection.VaultWatcherService. - Received a consumption event for a token E7499056A875F4C399DD30B4CB8C5D6346FBE8C21A9161449B4A985D0E26254A(0) which was either not present in the token cache, or was not locked {actor_id=user1, actor_owning_identity=O=Banorte, L=New York, C=US, actor_store_id=NODE_CONFIG, fiber-id=10000677, flow-id=367cc7c6-6a8c-4bf6-a56e-e2b02392c1ea, invocation_id=8fcf9e1e-8727-46a2-904b-7270eb10b3bd, invocation_timestamp=2019-12-09T03:33:16.504Z, origin=user1, session_id=271a869f-31f6-45c4-8610-42b1120753de, session_timestamp=2019-12-09T02:50:23.314Z, thread-id=144}

Which eventually would lead to a 

INFO ] 2019-12-09T02:00:22,092Z [Node thread-1] statemachine.StaffedFlowHospital. - Flow error allowed to propagate {actor_id=user1, actor_owning_identity=O=Banxico, L=London, C=GB, actor_store_id=NODE_CONFIG, fiber-id=10000008, flow-id=0c4f209b-84cc-47a6-9371-00b2e65dbf9c, invocation_id=f56a6cc9-580c-4 076-bb14-b5b3942d102b, invocation_timestamp=2019-12-09T02:00:13.770Z, origin=user1, session_id=e9b0bd3d-599a-474c-abf0-a2d557671d09, session_timestamp=2019-12-09T01:54:07.346Z, thread-id=145} java.lang.IllegalStateException: Insufficient spendable states identified for 0.01 TokenType(tokenIdentifier='MXN', fractionDigits=2)

**Comment by roger3cev** [comment]
Can you provide some more information about your set up please? 

The database token selection approach will never perform particularly well. Especially if there's contention over a small amount of tokens in the vault. This is why we added in memory token selection. 

Also, tokens 1.1 requires Corda 4.3.

Cheers

**Comment by jarturocastan** [comment]
Yes, I use corda 4.3, but this error happen when I try move token in many transactions, like 30 tx/s or many, sometimes less
